File: code_challenge_solution/java/37 Count words in a string.java
```java
public class WordCount {
    public static void main(String[] args) {
        String sentence = "Count words in a string";
        int wordCount = countWords(sentence);
        System.out.println("Word count: " + wordCount);
    }
    
    public static int countWords(String sentence) {
        if (sentence == null || sentence.isEmpty()) {
            return 0;
        }
        
        String[] words = sentence.split("\\s+");
        return words.length;
    }
}
```

File: code_challenge_solution/c++/37 Count words in a string.cpp
```
#include <iostream>
#include <string>
#include <sstream>

int countWords(std::string str) {
    std::stringstream ss(str);
    std::string word;
    int count = 0;
    
    while (ss >> word) {
        count++;
    }
    
    return count;
}

int main() {
    std::string sentence;
    
    std::cout << "Enter a sentence: ";
    std::getline(std::cin, sentence);
    
    int wordCount = countWords(sentence);
    
    std::cout << "Number of words in the sentence: " << wordCount << std::endl;
    
    return 0;
}
```